spec/plugin_spec.rb

```diff
@@ -1,55 +1,101 @@
 # frozen_string_literal: true
+
 require_relative '../lib/plugins/plugin.rb'
+require_relative '../lib/plugins/helpers/network.rb'
+require 'httparty'
 require 'spec_helper'
+require 'yaml'
 
 RSpec.describe Plugin, '#exec' do
-  context 'calls exec on plugins' do
+  before(:all) do
+    rackup = File.expand_path('../../bin/rackup', __FILE__)
+    @rack_pid = spawn("#{rackup} -D -E test -o 0.0.0.0 -p 4567", close_others: true)
```

Comment:
Eh, so I'm still not happy with this but it serves the following.
* Allows loading a bot that has an api endpoint available to test plugins to use the bot chat api
* Loads all the test plugins like we do with SLAPI and assigns ports that are open to each, ran into issues on travis with ports not being available that were locally
* Cleans up a lot of the test to just 2 lines, essential same test just different mock data and expected results, looking at shared examples for this in the future or something similar